### DISCUSSION TITLE
Requirements already satisfied...

### DISCUSSION THREAD

**Discussion by djptr** [adieyal/sd-dynamic-prompts#520] [Q&A] [answered]
Each time I start the WebUI, I get this list of 'requirements already satisfied' in the console. It's Dynamic Prompts related, but It looks like it didn't cleanup properly, and I don't know how to fix this. SD runs fine, and I just reinstalled the Dynamic Prompts extension with latest versions 2.12.5 & 0.27.0. but the list persists on each startup. What is causing this, and how can I fix it? Any help appreciated, thank you.

From the console:

Already up to date.
venv "C:\SD\stable-diffusion-webui\venv\Scripts\Python.exe"
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: v1.3.2
Commit hash: baf6946e06249c5af9851c60171692c44ef633e0
Installing requirements

Fetching updates for midas...
Checking out commit for midas with hash: 1645b7e...

Requirement already satisfied: send2trash~=1.8 in c:\sd\stable-diffusion-webui\venv\lib\site-packages (1.8.0)
Requirement already satisfied: dynamicprompts[attentiongrabber,magicprompt]~=0.27.0 in c:\sd\stable-diffusion-webui\venv\lib\site-packages (0.27.0)
Requirement already satisfied: jinja2~=3.1 in c:\sd\stable-diffusion-webui\venv\lib\site-packages (from dynamicprompts[attentiongrabber,magicprompt]~=0.27.0) (3.1.2)
Requirement already satisfied: pyparsing~=3.0 in c:\sd\stable-diffusion-webui\venv\lib\site-packages (from dynamicprompts[attentiongrabber,magicprompt]~=0.27.0) (3.0.9)
Requirement already satisfied: transformers[torch]~=4.19 in c:\sd\stable-diffusion-webui\venv\lib\site-packages (from dynamicprompts[attentiongrabber,magicprompt]~=0.27.0) (4.25.1)
Requirement already satisfied: MarkupSafe>=2.0 in c:\sd\stable-diffusion-webui\venv\lib\site-packages (from jinja2~=3.1->dynamicprompts[attentiongrabber,magicprompt]~=0.27.0) (2.1.2)
Requirement already satisfied: filelock in c:\sd\stable-diffusion-webui\venv\lib\site-packages (from transformers[torch]~=4.19->dynamicprompts[attentiongrabber,magicprompt]~=0.27.0) (3.9.0)
Requirement already satisfied: huggingface-hub<1.0,>=0.10.0 in c:\sd\stable-diffusion-webui\venv\lib\site-packages (from transformers[torch]~=4.19->dynamicprompts[attentiongrabber,magicprompt]~=0.27.0) (0.13.4)
Requirement already satisfied: numpy>=1.17 in c:\sd\stable-diffusion-webui\venv\lib\site-packages (from transformers[torch]~=4.19->dynamicprompts[attentiongrabber,magicprompt]~=0.27.0) (1.23.5)
Requirement already satisfied: packaging>=20.0 in c:\sd\stable-diffusion-webui\venv\lib\site-packages (from transformers[torch]~=4.19->dynamicprompts[attentiongrabber,magicprompt]~=0.27.0) (23.0)
Requirement already satisfied: pyyaml>=5.1 in c:\sd\stable-diffusion-webui\venv\lib\site-packages (from transformers[torch]~=4.19->dynamicprompts[attentiongrabber,magicprompt]~=0.27.0) (6.0)
Requirement already satisfied: regex!=2019.12.17 in c:\sd\stable-diffusion-webui\venv\lib\site-packages (from transformers[torch]~=4.19->dynamicprompts[attentiongrabber,magicprompt]~=0.27.0) (2022.10.31)
Requirement already satisfied: requests in c:\sd\stable-diffusion-webui\venv\lib\site-packages (from transformers[torch]~=4.19->dynamicprompts[attentiongrabber,magicprompt]~=0.27.0) (2.31.0)
Requirement already satisfied: tokenizers!=0.11.3,<0.14,>=0.11.1 in c:\sd\stable-diffusion-webui\venv\lib\site-packages (from transformers[torch]~=4.19->dynamicprompts[attentiongrabber,magicprompt]~=0.27.0) (0.13.2)
Requirement already satisfied: tqdm>=4.27 in c:\sd\stable-diffusion-webui\venv\lib\site-packages (from transformers[torch]~=4.19->dynamicprompts[attentiongrabber,magicprompt]~=0.27.0) (4.64.1)
Requirement already satisfied: torch!=1.12.0,>=1.7 in c:\sd\stable-diffusion-webui\venv\lib\site-packages (from transformers[torch]~=4.19->dynamicprompts[attentiongrabber,magicprompt]~=0.27.0) (2.0.0+cu118)
Requirement already satisfied: typing-extensions>=3.7.4.3 in c:\sd\stable-diffusion-webui\venv\lib\site-packages (from huggingface-hub<1.0,>=0.10.0->transformers[torch]~=4.19->dynamicprompts[attentiongrabber,magicprompt]~=0.27.0) (4.4.0)
Requirement already satisfied: sympy in c:\sd\stable-diffusion-webui\venv\lib\site-packages (from torch!=1.12.0,>=1.7->transformers[torch]~=4.19->dynamicprompts[attentiongrabber,magicprompt]~=0.27.0) (1.11.1)
Requirement already satisfied: networkx in c:\sd\stable-diffusion-webui\venv\lib\site-packages (from torch!=1.12.0,>=1.7->transformers[torch]~=4.19->dynamicprompts[attentiongrabber,magicprompt]~=0.27.0) (3.0)
Requirement already satisfied: colorama in c:\sd\stable-diffusion-webui\venv\lib\site-packages (from tqdm>=4.27->transformers[torch]~=4.19->dynamicprompts[attentiongrabber,magicprompt]~=0.27.0) (0.4.6)
Requirement already satisfied: charset-normalizer<4,>=2 in c:\sd\stable-diffusion-webui\venv\lib\site-packages (from requests->transformers[torch]~=4.19->dynamicprompts[attentiongrabber,magicprompt]~=0.27.0) (3.0.1)
Requirement already satisfied: idna<4,>=2.5 in c:\sd\stable-diffusion-webui\venv\lib\site-packages (from requests->transformers[torch]~=4.19->dynamicprompts[attentiongrabber,magicprompt]~=0.27.0) (2.10)
Requirement already satisfied: urllib3<3,>=1.21.1 in c:\sd\stable-diffusion-webui\venv\lib\site-packages (from requests->transformers[torch]~=4.19->dynamicprompts[attentiongrabber,magicprompt]~=0.27.0) (1.26.14)
Requirement already satisfied: certifi>=2017.4.17 in c:\sd\stable-diffusion-webui\venv\lib\site-packages (from requests->transformers[torch]~=4.19->dynamicprompts[attentiongrabber,magicprompt]~=0.27.0) (2022.12.7)
Requirement already satisfied: mpmath>=0.19 in c:\sd\stable-diffusion-webui\venv\lib\site-packages (from sympy->torch!=1.12.0,>=1.7->transformers[torch]~=4.19->dynamicprompts[attentiongrabber,magicprompt]~=0.27.0) (1.3.0)
sd-dynamic-prompts installer: running 'C:\SD\stable-diffusion-webui\venv\Scripts\python.exe' -m pip install 'send2trash~=1.8' 'dynamicprompts[attentiongrabber,magicprompt]~=0.27.0'

Launching Web UI with arguments: --xformers  
etc...

**Comment by akx** [comment]
Would be fixed by #509. For the time being, just ignore that.

**Reply by djptr** [reply]
Ok, thank you!

**Reply by djptr** [reply]
The issue persists I still have the 'requirements already satisfied' in the console on startup. I tried removing the Dynamic Prompts extension folder, and reinstalling it to no avail. I have no clue how to get rid of it...

**Reply by djptr** [reply, marked as answer]
The issue is fixed, the 'requirements already satisfied' in the console on startup are now gone. In case you fixed this, thank you!
